src/op_analytics/datapipeline/models/templates/aggregated_traces/tr_from_tr_to_hash.sql.j2

```diff
@@ -0,0 +1,153 @@
+  , tx.l2_priority_fee_native AS tx_l2_priority_fee_native
+  , tx.l2_base_fee_native AS tx_l2_base_fee_native
+  , tx.l2_legacy_extra_fee_native AS tx_l2_legacy_extra_fee_native
+
+  -- Transaction-Level Gas Prices
+  , tx.l2_gas_price_gwei AS tx_l2_gas_price_gwei
+  , tx.l2_base_gas_price_gwei AS tx_l2_base_gas_price_gwei
+  , tx.l2_priority_gas_price_gwei AS tx_l2_priority_gas_price_gwei
+  , tx.l2_legacy_extra_gas_price_gwei AS tx_l2_legacy_extra_gas_price_gwei
+  , tx.l1_base_gas_price_gwei AS tx_l1_base_gas_price_gwei
+  , tx.l1_blob_base_gas_price_gwei AS tx_l1_blob_base_gas_price_gwei
+
+  -- Transaction-Level transaction sizes
+  , tx.input_byte_length AS tx_input_byte_length
+  , tx.input_zero_bytes AS tx_input_zero_bytes
+  , (tx.input_byte_length - tx.input_zero_bytes) AS tx_input_nonzero_bytes
+  , tx.l1_base_scaled_size AS tx_l1_base_scaled_size
+  , tx.l1_blob_scaled_size AS tx_l1_blob_scaled_size
+  , tx.l1_gas_used_unified AS tx_l1_gas_used_unified
+  , tx.estimated_size AS tx_estimated_size
+
+  -- Attributes
+  , tx.success AS tx_success
+  , tx.is_system_transaction
+
+  -- Sum Internal Trace Gas
+  , SUM(tr.trace_gas_used) AS sum_trace_gas_used
+  , SUM(tr.gas_used_minus_subtraces) AS sum_trace_gas_used_minus_subtraces
+
+  , SUM(tr.tx_l2_fee_native_minus_subtraces) AS sum_tx_l2_fee_native_minus_subtraces
+  , SUM(tr.tx_l2_base_fee_native_minus_subtraces) AS sum_tx_l2_base_fee_native_minus_subtraces
+  , SUM(tr.tx_l2_priority_fee_native_minus_subtraces) AS sum_tx_l2_priority_fee_native_minus_subtraces
+  , SUM(tr.tx_l2_legacy_base_fee_native_minus_subtraces) AS sum_tx_l2_legacy_base_fee_native_minus_subtraces
+
+  , SUM(tr.tx_l2_gas_used_amortized_by_call) AS sum_tx_l2_gas_used_amortized_by_call
+  , SUM(tr.tx_l1_gas_used_unified_amortized_by_call) AS sum_tx_l1_gas_used_unified_amortized_by_call
+  , SUM(tr.tx_l1_base_scaled_size_amortized_by_call) AS sum_tx_l1_base_scaled_size_amortized_by_call
+  , SUM(tr.tx_l1_blob_scaled_size_amortized_by_call) AS sum_tx_l1_blob_scaled_size_amortized_by_call
+  , SUM(tr.tx_estimated_size_amortized_by_call) AS sum_tx_estimated_size_amortized_by_call
+
+  , SUM(tr.tx_fee_native_amortized_by_call) AS sum_tx_fee_native_amortized_by_call
+  , SUM(tr.tx_l2_fee_native_amortized_by_call) AS sum_tx_l2_fee_native_amortized_by_call
+  , SUM(tr.tx_l1_fee_native_amortized_by_call) AS sum_tx_l1_fee_native_amortized_by_call
+  , SUM(tr.tx_l2_base_fee_native_amortized_by_call) AS sum_tx_l2_base_fee_native_amortized_by_call
+  , SUM(tr.tx_l2_priority_fee_native_amortized_by_call) AS sum_tx_l2_priority_fee_native_amortized_by_call
+
+
+  , SUM(
+    (tx.l1_fee_native / tr.num_traces_in_txn)
+    + (tr.trace_gas_used - coalesce(tr.sum_subtraces_gas_used, 0)) * gwei_to_eth(tx.l2_gas_price_gwei)
+  ) AS sum_tx_fee_native_l1_amortized_l2_minus_subtraces
+
+
+  -- Count Call Types
+  , COUNT(*) FILTER (WHERE tr.trace_address = '' AND tr.call_type != '') AS count_top_level_calls
```

Comment:
Maybe validate that NULL trace addresses are represented as empty string in our db?